python/graphstorm/gconstruct/transform.py

```diff
@@ -1674,9 +1674,14 @@ def parse_label_ops(confs, is_node):
         custom_split = label_conf['custom_split_filenames']
         assert isinstance(custom_split, dict), \
                 "Custom data split needs to provide train/val/test index."
-        train_idx = read_index_json(custom_split['train']) if 'train' in custom_split else None
-        val_idx = read_index_json(custom_split['valid']) if 'valid' in custom_split else None
-        test_idx = read_index_json(custom_split['test']) if 'test' in custom_split else None
+        if "column" not in custom_split:
+            custom_split["column"] = []
+        train_idx = read_index(custom_split['train'], custom_split["column"]) \
```

Comment:
How about `train_idx = read_index(custom_split_info)` and hide the complexity inside read_index?

Reply:
That is a good idea.